embeddings/model/lightning_model.py

```diff
@@ -13,17 +15,21 @@ def __init__(
         self,
         trainer: pl.Trainer,
         task: HuggingFaceLightningTask,
-        predict_subset: Literal["dev", "test"] = "test",
+        predict_subset: LightingDataModuleSubset,
     ) -> None:
         super().__init__()
         self.trainer = trainer
         self.task = task
         self.predict_subset = predict_subset
 
     def execute(self, data: pl.LightningDataModule) -> Dict[str, nptyping.NDArray[Any]]:
-        self.trainer.fit(self.task, data)
-        dataloader = (
-            data.test_dataloader() if self.predict_subset == "test" else data.val_dataloader()
-        )
+        try:
+            self.trainer.fit(self.task, data)
+        except Exception as e:
+            del self.trainer
+            torch.cuda.empty_cache()  # type: ignore
+            raise e
+
+        dataloader = get_subset_from_lighting_datamodule(data=data, subset=self.predict_subset)
```

Comment:
maybe move this functionality to datamodule? 
```suggestion
        dataloader = data.get_subset(self.predict_subset)
```

Reply:
Done